protocol/xmrmaker/swap_state.go

```diff
@@ -40,9 +40,9 @@ var (
 type swapState struct {
 	backend.Backend
 
-	ctx    context.Context
-	cancel context.CancelFunc
-	sync.Mutex
```

Comment:
I changed the swapState<->lock relationship from `is-a` to a `has-a`, because without a variable the code is hard to audit, as you can't search for everywhere the lock is used.  You can search on the `Lock` and `Unlock` methods, but then you'll end up finding every lock use, not the use on a specific maker or taker swapState.